src/libraries/System.Text.RegularExpressions/ref/System.Text.RegularExpressions.cs

```diff
@@ -287,4 +289,46 @@ public abstract partial class RegexRunnerFactory
         protected RegexRunnerFactory() { }
         protected internal abstract System.Text.RegularExpressions.RegexRunner CreateInstance();
     }
+    public enum RegexParseError
+    {
+        Unknown,
+        AlternationHasTooManyConditions,
+        AlternationHasMalformedCondition,
+        InvalidUnicodePropertyEscape,
+        MalformedUnicodePropertyEscape,
+        UnrecognizedEscape,
+        UnrecognizedControlCharacter,
+        MissingControlCharacter,
+        InsufficientOrInvalidHexDigits,
+        QuantifierOrCaptureGroupOutOfRange,
+        UndefinedNamedReference,
+        UndefinedNumberedReference,
+        MalformedNamedReference,
+        UnescapedEndingBackslash,
+        UnterminatedComment,
+        InvalidGroupingConstruct,
+        AlternationHasNamedCapture,
+        AlternationHasComment,
+        AlternationHasMalformedReference,
+        AlternationHasUndefinedReference,
+        CaptureGroupNameInvalid,
+        CaptureGroupOfZero,
+        UnterminatedBracket,
+        ExclusionGroupNotLast,
+        ReversedCharacterRange,
+        ShorthandClassInCharacterRange,
+        InsufficientClosingParentheses,
+        ReversedQuantifierRange,
+        NestedQuantifiersNotParenthesized,
+        QuantifierAfterNothing,
+        InsufficientOpeningParentheses,
+        UnrecognizedUnicodeProperty
+    }
+    public sealed class RegexParseException : ArgumentException
+    {
+        public RegexParseException(RegexParseError error, int offset) { }
```

Comment:
Is this the accepted design?  The exception is so specific that we don't want or need the standard ctors that most other exceptions have?

Reply:
Indeed, it was specifically designed to disallow overriding the message, because the main use case here is to allow more control on catching errors, not generating your own. For the same reason, they didn't want to make the class inheritable.

These decisions also had something to do with backwards compat: it has to be binary compatible with ArgumentException.

Reply:
Who uses this ctor?

Reply:
@stephentoub, the public ctor is probably not going to be used much, if at all, but it was requested by the api review team. @danmosemsft mentioned that if we want to change the signature (which I did at some point in the commits), then it has to go through the review process again.

Bottom line in the discussion above was that we keep it in and create a default message string, which btw was also what was suggested in the video review. 

Internally, there's no code that calls this. I think we should just leave it as is.

Reply:
It's considered poor form to expose exceptions that don't let you configure the message or inner exception. There are design guidelines for that. So I'm surprised this made it through API review like this.
cc: @bartonjs